components/raftstore/src/store/peer.rs

```diff
@@ -2380,6 +2397,26 @@ impl Peer {
 
         poll_ctx.raft_metrics.propose.normal += 1;
 
+        // If applied index's term is differ from current raft's term, leader transfer
+        // must happened.
+        if self.get_store().applied_index_term() != self.term() {
```

Comment:
Here is to ensure this leader has applied all proposals belongs to the previous leaders. Otherwise, the epoch checker may be broken because of the lack of epoch information.

Reply:
Oh. The annotation is wrong. I copy them from another place...